jobs/blobstore/templates/read_users.erb

```diff
@@ -1,9 +1,12 @@
 <%=
 all_users = {
-	p('blobstore.agent.user') => p('blobstore.agent.password'),
-	p('blobstore.director.user') => p('blobstore.director.password'),
+  p('blobstore.director.user') => p('blobstore.director.password'),
 }
 
+if_p('blobstore.agent.user') do
+  all_users[p('blobstore.agent.user')] = p('blobstore.agent.password')
+end
```

Comment:
I thought that we want to make these properties depend on `blobstore.enable_signed_urls`?

Reply:
If `blobstore.agent.user` is not set and `blobstore.enable_signed_urls` is false it should raise an error. In order to shorten the operator feedback loop in the case of a misconfiguration.

Reply:
I think we shouldn't configure the `blobstore.agent.user` if `blobstore.enable_signed_urls` is set to true.
The agent doesn't need this credential anymore in this case. And from a security perspective, it's better to not configure it so that you don't need to rotate it.

We currently have the issue that some CPIs still reference the `blobstore.agent.user`, `blobstore.agent.password` even though they don't use it if `blobstore.enable_signed_urls` is set to true.

Till these CPIs have been updated, we could set blobstore.agent.user`, `blobstore.agent.password` in the manifest and still wouldn't need to rotate the password, since it is not configured in the blobstore.

Reply:
When mixing old and new stemcells it could in theory be that both signed URLs and non agent blobstore credentials are used at the same time: https://github.com/cloudfoundry/bosh/blob/master/src/bosh-director/lib/bosh/blobstore_client/base.rb#L122-L125

Apparently the ubuntu-xenial/456.x line (which is still supported for VMware customers) does not support singed URLs ([API version 2](https://github.com/cloudfoundry/bosh-linux-stemcell-builder/blob/ubuntu-xenial/456.x/bosh-stemcell/lib/bosh/stemcell/stemcell_packager.rb#L45)). 
Stemcell builder was bumped to API version 3 in this commit: https://github.com/cloudfoundry/bosh-linux-stemcell-builder/commit/f4de1ec001c4dd653f24d7ba52efa6a31f139e19